examples/vision/cutmix.py

```diff
@@ -0,0 +1,367 @@
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.title(class_names[np.argmax(label_batch[i])])
+    plt.imshow(image_batch[i])
+    plt.axis("off")
+
+"""
+## Define the model using ResNet-20
+"""
+
+
+def resnet_layer(
+    inputs,
+    num_filters=16,
+    kernel_size=3,
+    strides=1,
+    activation="relu",
+    batch_normalization=True,
+    conv_first=True,
+):
+
+    conv = keras.layers.Conv2D(
+        num_filters,
+        kernel_size=kernel_size,
+        strides=strides,
+        padding="same",
+        kernel_initializer="he_normal",
+        kernel_regularizer=keras.regularizers.l2(1e-4),
+    )
+
+    x = inputs
+    if conv_first:
+        x = conv(x)
+        if batch_normalization:
+            x = keras.layers.BatchNormalization()(x)
+        if activation is not None:
+            x = keras.layers.Activation(activation)(x)
+    else:
+        if batch_normalization:
+            x = keras.layers.BatchNormalization()(x)
+        if activation is not None:
+            x = keras.layers.Activation(activation)(x)
+        x = conv(x)
+    return x
+
+
+def resnet_v20(input_shape, depth, num_classes=10):
+
+    if (depth - 2) % 6 != 0:
+        raise ValueError("depth should be 6n+2 (eg 20, 32, 44 in [a])")
+    # Start model definition.
+    num_filters = 16
+    num_res_blocks = int((depth - 2) / 6)
+
+    inputs = keras.layers.Input(shape=input_shape)
+    x = resnet_layer(inputs=inputs)
+    # Instantiate the stack of residual units
+    for stack in range(3):
+        for res_block in range(num_res_blocks):
+            strides = 1
+            if stack > 0 and res_block == 0:  # first layer but not first stack
+                strides = 2  # downsample
+            y = resnet_layer(inputs=x, num_filters=num_filters, strides=strides)
+            y = resnet_layer(inputs=y, num_filters=num_filters, activation=None)
+            if stack > 0 and res_block == 0:  # first layer but not first stack
+                # linear projection residual shortcut connection to match
+                # changed dims
+                x = resnet_layer(
+                    inputs=x,
+                    num_filters=num_filters,
+                    kernel_size=1,
+                    strides=strides,
+                    activation=None,
+                    batch_normalization=False,
+                )
+            x = keras.layers.add([x, y])
+            x = keras.layers.Activation("relu")(x)
+        num_filters *= 2
+
+    # Add classifier on top.
+    # v1 does not use BN after last shortcut connection-ReLU
+    x = keras.layers.AveragePooling2D(pool_size=8)(x)
+    y = keras.layers.Flatten()(x)
+    outputs = keras.layers.Dense(
+        num_classes, activation="softmax", kernel_initializer="he_normal"
+    )(y)
+
+    # Instantiate model.
+    model = keras.models.Model(inputs=inputs, outputs=outputs)
+    return model
+
+
+def training_model():
+    return resnet_v20((32, 32, 3), 20)
```

Comment:
Note: in practice you get much better accuracy on CIFAR-10 by upscaling the input images (perhaps this is what you saw better accuracy with ResNet-20...), i.e. by using a (None, None, 3) as input shape and starting the model with a `Rescaling(150, 150)` layer. Have you tried that?

Reply:
No haven't tried that should I do try that?